docs/Halogen/Menu/Model.md

```diff
@@ -0,0 +1,21 @@
+## Module Halogen.Menu.Model
+
+#### `Menu`
+
+``` purescript
+type Menu a = { chosen :: Maybe Int, submenus :: Array (MenuItem a) }
+```
+
+#### `MenuItem`
+
+``` purescript
+type MenuItem a = { label :: String, submenu :: Submenu a }
+```
+
+#### `mkMenu`
+
+``` purescript
+mkMenu :: forall a. Array (MenuItem a) -> Menu a
```

Comment:
How do you feel about calling this `makeMenu` instead of `mkMenu`? (I don't really have a strong preference, but I tend to default to writing the full word out.)